src/mruset.h

```diff
@@ -22,21 +22,26 @@ class mruset
 
 protected:
     std::set<T> set;
-    std::deque<T> queue;
+    std::vector<iterator> order;
+    size_type first_used;
+    size_type first_unused;
     size_type nMaxSize;
 
 public:
-    mruset(size_type nMaxSizeIn = 0) { nMaxSize = nMaxSizeIn; }
+    mruset(size_type nMaxSizeIn = 1) { clear(nMaxSizeIn); }
     iterator begin() const { return set.begin(); }
     iterator end() const { return set.end(); }
     size_type size() const { return set.size(); }
     bool empty() const { return set.empty(); }
     iterator find(const key_type& k) const { return set.find(k); }
     size_type count(const key_type& k) const { return set.count(k); }
-    void clear()
+    void clear(size_type nMaxSizeIn = 1)
     {
         set.clear();
-        queue.clear();
+        std::vector<iterator>(nMaxSizeIn, set.end()).swap(order);
```

Comment:
It's horrible, but it's the only way to make a vector release its memory, afaik.